packages/frontend/src/retro/hooks/useTimer.ts

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState } from "react";
+
+interface useTimerProps {
+  onTimerFinish: () => void;
+  runtime: number;
+}
+export function useTimer({ onTimerFinish, runtime }: useTimerProps) {
+  const [isRunning, setIsRunning] = useState(false);
+  const [isPaused, setIsPaused] = useState(false);
+  const [remainingTime, setRemainingTime] = useState(runtime);
+
+  function startTimer(duration: number) {
+    console.log("Start Timer at ", new Date().toISOString());
+    // setRemainingTime(finishTime - Date.now());
+    setRemainingTime(duration);
+    setIsRunning(true);
+  }
+
+  function stopTimer(isFinished: boolean = false) {
+    console.log("Stop timer at ", new Date().toISOString());
+    setIsRunning(false);
+    setIsPaused(false);
+    if (isFinished) {
+      onTimerFinish();
+    }
+  }
+
+  function pauseTimer() {
+    console.log("pause timer");
+    setIsPaused(() => !isPaused);
+  }
+
+  function createLabel() {
+    return new Date(remainingTime).toISOString().slice(14, 19);
+  }
+
+  function getMinutes(ms: number) {
+    if (ms < 60000) {
+      return 0;
+    }
+    // console.log("minutes {}", Math.floor(ms / 1000 / 60));
+    return Math.floor(ms / 1000 / 60);
+  }
+
+  function getSeconds(ms: number) {
+    // console.log("seconds {}", Math.round((ms / 1000) % 60));
+    return Math.round((ms / 1000) % 60);
+  }
+
+  useEffect(() => {
+    function updateTimer() {
+      setRemainingTime(() => Math.max(remainingTime - 1000, 0));
```

Comment:
Using the callback function in setState is useful when the new state depends oh the old state. In that case, the callback function has a parameter that can be used here instead of remainingTime